### PR TITLE
Fixes #36088 - translate the description values on the settings page

### PR DESCRIPTION
**Pull request by nofaralfasi**
Add the translatedCellFormatter.js file, which
enables the translation of settings descriptions.

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import React from 'react';
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js]: There's no React component in this formatter, so you could probably remove this line and use the formatter as a regular function

File: webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+import cellFormatter from './cellFormatter';
+
+export default value => cellFormatter(__(value));
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js:5]: strings are not really captured for translation on runtime,
you need to mark an actual string, not a variable in order to mark it for translation,
you can read more about the process in https://projects.theforeman.org/projects/foreman/wiki/Translating

File: webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { sprintf, translate as __ } from '../../../../../react_app/common/I18n';
+import cellFormatter from './cellFormatter';
+
+export default value => cellFormatter(sprintf(__('%s'), value));
```
Comment (adamruzicka) [inline webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js]: How should this work?

I'd expect that
```javascript
> __('%s') == '%s'
true

> sprintf('%s', 'thingy') == 'thingy'
true
```

unless I'm missing something this feels like a fancy identity function

File: webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js (hunk 2 above)
Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js:5]: Does that matter in this case? Aren't the strings to be displayed loaded from the backend where they (hopefully) should be marked? So in this case it indeed couldn't be used to extract strings, but since they are extracted elsewhere i wouldn't matter and they would just get translated?

File: webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js (hunk 3 above)
Comment (nofaralfasi) [reply webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js]: I tried to follow the guide Ron mentioned [here](https://github.com/theforeman/foreman/pull/9713#discussion_r1199751518).
And I found some examples with similar usages (for [example](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/common/forms/validators.js#L5 ))

Comment (ekohl) [reply webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js]: The string extraction process is relatively dumb. It simply parses the code and looks at what's between `__()`. So this will result in it finding a string `%s`. Then it interpolates it (using `sprintf`). However, it will never find the content of `value`. So you can also write (as you did earlier) `__(value)`.

But note that you should make sure that the real strings get extracted somewhere. You can imagine that finding out which content `value` could ever have is a near impossible task. In this particular case it's already taken care of because it happens in the setting declaration, but that's not always the case if you use translations.

File: webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js (hunk 2 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js:5]: if they are translated already in the backend,  I believe we should just display them in the table without any extra translation/marking.

Comment (ekohl) [reply webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js:5]: The descriptions are in files such as https://github.com/theforeman/foreman/blob/develop/app/registries/foreman/settings/general.rb and you can see they are marked for translation (using `N_()`, see https://projects.theforeman.org/projects/foreman/wiki/Translating#Extracting-strings):
https://github.com/theforeman/foreman/blob/f7e5525dc8a3af7dd5333038f633f4e8b462f5be/app/registries/foreman/settings/general.rb#L8

That means they are present in the extracted strings, but you need to choose when to translate them. I don't even think the API can translate them. Perhaps that would be a good thing, but will be a far larger effort and we already have all the translations present in the frontend code.

In short: the original `value => cellFormatter(__(value))` looks correct to me, provided there is some asterisk that the caller must ensure the input is extracted somewhere. I don't know what a good way is, since I don't know how we're supposed to document the frontend code, but the least is a code comment I guess.

File: webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js (hunk 3 above)
Comment (nofaralfasi) [reply webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js]: Thanks @ekohl!

**Review by ekohl — APPROVED** [review]
I didn't test this and don't know JS well enough to comment on the specifics, but from a translation point of view I think this looks correct.

File: webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js (hunk 2 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/common/table/formatters/translatedCellFormatter.js:5]: well, I learned something new, I though we use the translations only for strings nor variables,
and in such case the server should pass them to the frontend already translated.
but in this case it seem to actually work 👍🏼

**Review by Ron-Lavi — APPROVED** [review]
Thanks @nofaralfasi ! fixed the issue for me
